src/coreclr/vm/stublink.h

```diff
@@ -784,21 +830,17 @@ class Stub
         };
 
         UINT32  m_signature;
-#else
 #ifdef HOST_64BIT
-        //README ALIGNMENT: in retail mode UINT m_numCodeBytes does not align to 16byte for the code
-        //                   after the Stub struct. This is to pad properly
-        UINT    m_pad_code_bytes;
+        //README ALIGNMENT: Enusure code after the Stub struct align to 16-bytes.
+        UINT32  m_pad_code_bytes1;
+        UINT32  m_pad_code_bytes2;
+        UINT32  m_pad_code_bytes3;
 #endif // HOST_64BIT
 #endif // _DEBUG
 
-#ifdef _DEBUG
-        Stub()      // Stubs are created by NewStub(), not "new". Hide the
-        { LIMITED_METHOD_CONTRACT; }          //  constructor to enforce this.
-#endif
-
+        Stub() = delete; // Stubs are created by NewStub(), not "new".
 };
-
+static_assert_no_msg(sizeof(Stub) % CODE_SIZE_ALIGN == 0);
```

Comment:
This gets defined in cgencpu, and that one is supposed to be the one to include this file.

Reply:
Maybe we can move this to the cpp file?

Reply:
Too many levels of indirection in that statement. What is "This" and "that one" mean?

Reply:
Sorry, bad habit. Compilation errors are due to an undefined `CODE_SIZE_ALIGN`. That gets defined in `cgencpu` which is the file that's responsible for including this header. You can't really use it here, I think the easiest fix is to move the assert to stublink.cpp